code/modules/mob/living/carbon/human/human_attackhand.dm

```diff
@@ -181,10 +181,21 @@
 		visible_message("<span class='danger'>[M] attempts to touch [src]!</span>")
 		return 0
 
-
-	if(istype(M,/mob/living/carbon))
-//		log_debug("No gloves, [M] is truing to infect [src]")
-		spread_disease_to(M, src, "Contact")
+	var/touch_zone = FULL_TORSO
+	var/datum/organ/external/S = src.get_organ(M.zone_sel.selecting)
+	if(M.zone_sel.selecting == "head" && !(!S || S.status & ORGAN_DESTROYED))
```

Comment:
Maybe check for `S.status` and stuff and then use a `switch` construct on `zone_sel.selecting`